app.js

```diff
@@ -0,0 +1,184 @@
+import express from 'express';
+import crypto from 'crypto';
+import { products } from './storage.js';
+import { carts } from './storage.js';
+import { users } from './storage.js';
+import { orders } from './storage.js';
+
+const app = express();
+const PORT = 3000;
+
+app.use(express.json());
+
+// register/create user
+app.post('/api/register', (req, res) => {
+    const { email, name, password } = req.body;
+
+    // unique user id is generated
+    const userId = crypto.randomUUID();
+
+    //regx email check user@example.com 
+    const correctEmail = /^[a-zA-Z0-9.-_]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+    
+    //check whether email format is correct. If it is wrong, return error;
+    if(!correctEmail.test(email)) {
+        return res.status(400).json({ error: 'Invalid email format' });
+    }
+
+    //check length of the email
+    if (email.length > 254) {
+        return res.status(400).json({ error: 'Email is too long (max 254 characters)' });
+    }
+
+    const correctPassword = /^(?=.*[A-Z])(?=.*[a-z])(?=.*\d)(?=.*[!@#$%^&*])[A-Za-z\d!@#$%^&*]{8,}$/;
+
+    //check the password
+    if(!correctPassword.test(password)) {
+        return res.status(400).json({ error: 'Invalid email format' });
+    }
+
+    if (!email || !name || !password) {
```

Comment:
It would be better to check if these fields exist earlier, because if there is no email then there is no sense in performing other checks

Reply:
Thank you, changed the position of check.